src/iplant_groups/service/groups.clj

```diff
@@ -35,12 +36,14 @@
 
 (defn update-group-privileges
   [group-name {:keys [updates]} {:keys [user replace] :or {replace true} :as params}]
+  (when replace (util/verify-not-removing-own-privileges user (map :subject_id updates)))
   (doseq [[privileges vs] (group-by (comp set :privileges) updates)]
     (grouper/update-group-privileges user replace group-name (mapv :subject_id vs) privileges))
   (get-group-privileges group-name params))
 
 (defn remove-group-privileges
   [group-name {:keys [updates]} {:keys [user] :as params}]
+  (util/verify-not-removing-own-privileges user (map :subject_id updates))
```

Comment:
Yeah, but I'm going to have to update terrain for it. I've got a code change ready to go.